### ISSUE TITLE
[feat]: Drag n Drop

### ISSUE THREAD

**Issue by Jay-Karia** [shadcn-ui/ui#4629]
### Feature description

I would like to request a new component: Drag and Drop.

### Affected component/components

N/A

### Additional Context

Add a drag and drop lists which might be helpful in todo lists.

Similar to React DND

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by Kumar06Lav** [comment]
Hi team, I'd like to take on this issue. Could you please assign it to me? Thanks!

**Comment by Jay-Karia** [comment]
@Kumar06Lav no need to assign, you can send a pull supporting it

**Comment by jdgamble555** [comment]
This would be awesome!

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to two years of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you.

**Comment by jdgamble555** [comment]
Not stale, how is there 2 years of inactivity when this was opened less than a year ago!?

This would be cool to use  [Sortable](https://github.com/SortableJS/Sortable) for drag and drop!

J

**Comment by shadcn** [comment]
Yeah. There was an issue with the stale bot. Not stale. I removed the label.

**Comment by Krish-Das** [comment]
## Proposal: New Drag and Drop Component

Hi team 👋

I’ve been exploring the idea of introducing a **Drag and Drop** component for `shadcn/ui`.
This would enable reorderable lists and sortable layouts (not limited), particularly useful for use cases such as to-do lists or Kanban-style interfaces.

Before I start drafting a PR, I’d like to confirm whether the team is **open to including** such a component in the library.

---

### Prototype Preview
I’ve built a working prototype that demonstrates the interaction.  
For now, here’s a short preview recording:

https://github.com/user-attachments/assets/6ad381c7-ae4f-40d7-83fc-221c188688cd

---

### Key Features
- Restricts drops to valid positions (prevents dropping in the same place)
- Visual **hitbox indicator** showing where the item will land
- Custom **drag preview**
- **High performance**, powered by native browser APIs (uses external DnD libraries)
- **Handle support** (can be restricted to drag handles)
- **Minimal bundle size**

---

If this direction aligns with the project’s roadmap, I’d be happy to formalize it into a PR proposal with the API design and code.

Thanks for your time, and I’d appreciate your feedback before proceeding.
